torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -242,9 +242,48 @@ Value* TracingState::getOutput(const IValue& iv, size_t i) {
         fmap(tuple, [&](const IValue& ival) { return getOutput(ival, i); }));
     graph->insertNode(tuple_node);
     return tuple_node->output();
+  } else if (iv.isGenericDict()) {
+    if (tracing_mode_strict) {
+      throw std::runtime_error(
+          "Encountering a dict at the output of the tracer" +
+          std::string(STRICT_TRACER_MSG));
+    }
+    auto dict = iv.toGenericDict();
+    TypePtr key_type = dict.keyType();
+    TypePtr value_type = dict.valueType();
+
+    if (!(key_type->cast<StringType>() || key_type->cast<TensorType>()) ||
```

Comment:
This is kinda hard to read in the !(A || B) || !C form. Can we do something like:
```
bool key_type_valid = key_type->cast<StringType>() || key_type->cast<TensorType>();
bool value_type_valid = value_type->cast<TensorType>();

if (!key_type_valid || !value_type_valid) {
  <...>
```